### PR TITLE
Improve the documentation of PageElement's elementId

### PR DESCRIPTION
**Pull request by shtukas**
## What does this change?

We had a problem with videos display which was solved here: https://github.com/guardian/frontend/pull/24713 , and in the process of solving it the PageElement's elementId documentation came to the forefront. This PR makes it better (remove spelling mistakes and improve the formulation).

### CODE REVIEW

File: common/app/model/dotcomrendering/PageElement-Identifiers.md [hunk 1]
```diff
@@ -23,11 +25,19 @@ case class MediaAtomBlockElement(
     ...
 ```
 
-but this `id` should be seen as metadata about the original data the PageElement represents. 
+but this `id`, as mentionned above, a CAPI id, and should be seen as metadata about the original data that the PageElement represents. 
+
+The general situation is that not all `BlockElement`s from CAPI have a naturally defined unique identifier. Moreover, `PageElement`s are more general than content held in CAPI. 
 
-In Feb 2021, the DCR team requested that PageElements be given an identifier that would help with rendering, notably to be used for DOM reconciliation during React rendering. Although DCR itself could be responsible to provide them (ultil this point it actually did using integer indices), the backend accepted to provide one, called `elementId`. 
+(Indeed, as much as there should be a 121 correspondance between `PageElement`s in the backend, `BlockElement`s in the DCR data model, and ideally, DCR React Components, there is no direct correspondance between CAPI elements and `PageElement`s. This decision was taken by the Dotcom team at the time to free the design of DCR from inefficiencies in the CAPI data model.)
```
Comment (mxdvl) [inline common/app/model/dotcomrendering/PageElement-Identifiers.md:32]:
```suggestion
(Indeed, as much as there should be a 1-to-1 correspondence between `PageElement`s in the backend, `BlockElement`s in the DCR data model, and ideally, DCR React Components, there is no direct correspondence between CAPI elements and `PageElement`s. This decision was taken by the Dotcom team at the time to free the design of DCR from inefficiencies in the CAPI data model.)
```

File: common/app/model/dotcomrendering/PageElement-Identifiers.md [hunk 2]
```diff
@@ -48,12 +58,11 @@ to this:
...
 
-2. The initial implementation uses UUIDs, but the contract is that any reasonably unique string can do 
+2. The initial implementation uses UUIDs, but in essence any reasonably unique string can do. 
 
-3. The value of `elementId` for each element is, as per original implementation, randomly chosen at each generation. In any case, there is no 1-2-1 mapping between `PageElement`s / `BlockElement`s and those values. 
+3. The value of `elementId` for each element is, as per original implementation, randomly chosen at each generation of the DCR data object. In any case, there is no 1-2-1 mapping between `PageElement`s or `BlockElement`s and those values.
```
Comment (mxdvl) [inline common/app/model/dotcomrendering/PageElement-Identifiers.md:65]:
```suggestion
3. The value of `elementId` for each element is, as per original implementation, randomly chosen at each generation of the DCR data object. In any case, there is no 1-to-1 mapping between `PageElement`s or `BlockElement`s and those values. 
```

**Review by mxdvl — APPROVED** [review]
small notes for consistency

**Comment by shtukas** [comment]
Amazing thanks @mxdvl

File: common/app/model/dotcomrendering/PageElement-Identifiers.md [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
+Q: What is this id? A: This document is mentionned in a couple of comments in the frontend source code. The comments mention that one should look up that id in a string search to find it. 
+
+## PageElement Identifiers
+
+This document explains why `elementId` was introduced as attribute of `PageElement`s and DCR's `BlockElement`s
 
 ### Introduction
 
-`trait PageElement` and its members model the objects the backend send to DCR for rendering. In the DCR data object, they are referred to as `BlockElement`s, for instance 
+`trait PageElement` models the objects that the backend ("backend" here refers to the frontend Scala code) sends to DCR for rendering. In the DCR Data Object (the JSON object sent to DCR), they are referred to as `BlockElement`s. For instance
```
Comment (arelra) [inline common/app/model/dotcomrendering/PageElement-Identifiers.md:11]:
```suggestion
`trait PageElement` models the objects that the backend ("backend" here refers to Frontend's backend Scala application) sends to DCR for rendering. In the DCR Data Object (the JSON object sent to DCR), they are referred to as `BlockElement`s. For instance 
```

**Review by arelra — COMMENTED** [review]
Thanks for updating! Just some very minor suggestions 👍

**Review by arelra — APPROVED** [review]

**Pull request merged**